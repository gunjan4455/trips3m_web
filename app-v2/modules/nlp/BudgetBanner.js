import  React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import BudgetRange from 'app-v2/modules/nlp/BudgetRange';
import { CloseIconWhite } from 'helpers/Icon/Icon';

import styles from './BudgetBanner.cm.scss';


class BudgetBanner extends PureComponent {
  render() {
    const { destination, priceMax, priceMin, savePreferencesAndRedirect, adult, child } = this.props;
    return (
      <div className="at_nurturingBanner relative sbcw pb32">
        <span className={`absolute t15 l15 ${styles.nurtureIcon}`} onClick={() => savePreferencesAndRedirect()}>
          <CloseIconWhite />
        </span>
        <div className={`at_nurturingDiv pbc5 flex alignCenter justifyCenter ${styles.bannerHeading}`}>
          <p className="f24 sfcw fw9 text-center ">Your Total Budget for<br /> {destination} will be</p>
        </div>
        <div className={`absolute wfull ${styles.budgetWrap}`}>
          <BudgetRange priceMax={priceMax} priceMin={priceMin} />
          <div className={`absolute text-center p10 f12 sbc7 ${styles.numberOfPeople}`}>
            for {adult} {+adult > 1 ? 'adults' : 'adult'} {+child ? `& ${child} ${+child > 1 ? 'children' : 'child'} ` : null}
          </div>
        </div>
      </div>
    );
  }
}

BudgetBanner.propTypes = {
  destination: PropTypes.string.isRequired,
  priceMax: PropTypes.number,
  priceMin: PropTypes.number,
  savePreferencesAndRedirect: PropTypes.func.isRequired,
  adult: PropTypes.string,
  child: PropTypes.string
};

BudgetBanner.defaultProps = {
  adult: '2',
  child: '0'
};

export default BudgetBanner;

