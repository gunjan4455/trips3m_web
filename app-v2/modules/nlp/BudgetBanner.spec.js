import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import BudgetBanner from './BudgetBanner';

configure({ adapter: new Adapter() });

const DEFAULT_PROPS = {
  "destination": "Kerala",
  "priceMax": 142767,
  "priceMin": 55152,
  "savePreferencesAndRedirect": () => {
  }
}

const shallowWrapper = (props) => {
  return shallow(<BudgetBanner {...props} />);
}

describe('BudgetBanner Module', () => {
  let props = { ...DEFAULT_PROPS };

  it('should render correctly', () => {
    const wrapper = shallowWrapper(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correct destination', () => {
    const wrapper = shallowWrapper(props);
    expect(wrapper.find('.at_nurturingDiv').text()).toMatch(/Kerala/);
  });
});
