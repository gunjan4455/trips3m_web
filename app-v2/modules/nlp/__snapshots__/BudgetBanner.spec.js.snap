// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BudgetBanner Module should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <BudgetBanner
    adult={1}
    child={0}
    destination="Kerala"
    priceMax={142767}
    priceMin={55152}
    savePreferencesAndRedirect={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="absolute t15 l15 nurtureIcon"
          onClick={[Function]}
        >
          <IconSvg />
        </span>,
        <div
          className="at_nurturingDiv pbc5 flex alignCenter justifyCenter bannerHeading"
        >
          <p
            className="f24 sfcw fw9 text-center "
          >
            Your Total Budget for
            <br />
             
            Kerala
             will be
          </p>
        </div>,
        <div
          className="absolute wfull budgetWrap"
        >
          <BudgetRange
            priceMax={142767}
            priceMin={55152}
          />
          <div
            className="absolute text-center p10 f12 sbc7 numberOfPeople"
          >
            for 
            1
             
            adult
          </div>
        </div>,
      ],
      "className": "at_nurturingBanner relative sbcw pb32",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <IconSvg />,
          "className": "absolute t15 l15 nurtureIcon",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p
            className="f24 sfcw fw9 text-center "
          >
            Your Total Budget for
            <br />
             
            Kerala
             will be
          </p>,
          "className": "at_nurturingDiv pbc5 flex alignCenter justifyCenter bannerHeading",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Your Total Budget for",
              <br />,
              " ",
              "Kerala",
              " will be",
            ],
            "className": "f24 sfcw fw9 text-center ",
          },
          "ref": null,
          "rendered": Array [
            "Your Total Budget for",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            " ",
            "Kerala",
            " will be",
          ],
          "type": "p",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <BudgetRange
              priceMax={142767}
              priceMin={55152}
            />,
            <div
              className="absolute text-center p10 f12 sbc7 numberOfPeople"
            >
              for 
              1
               
              adult
            </div>,
          ],
          "className": "absolute wfull budgetWrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "priceMax": 142767,
              "priceMin": 55152,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "for ",
                1,
                " ",
                "adult",
                null,
              ],
              "className": "absolute text-center p10 f12 sbc7 numberOfPeople",
            },
            "ref": null,
            "rendered": Array [
              "for ",
              1,
              " ",
              "adult",
              null,
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="absolute t15 l15 nurtureIcon"
            onClick={[Function]}
          >
            <IconSvg />
          </span>,
          <div
            className="at_nurturingDiv pbc5 flex alignCenter justifyCenter bannerHeading"
          >
            <p
              className="f24 sfcw fw9 text-center "
            >
              Your Total Budget for
              <br />
               
              Kerala
               will be
            </p>
          </div>,
          <div
            className="absolute wfull budgetWrap"
          >
            <BudgetRange
              priceMax={142767}
              priceMin={55152}
            />
            <div
              className="absolute text-center p10 f12 sbc7 numberOfPeople"
            >
              for 
              1
               
              adult
            </div>
          </div>,
        ],
        "className": "at_nurturingBanner relative sbcw pb32",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <IconSvg />,
            "className": "absolute t15 l15 nurtureIcon",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p
              className="f24 sfcw fw9 text-center "
            >
              Your Total Budget for
              <br />
               
              Kerala
               will be
            </p>,
            "className": "at_nurturingDiv pbc5 flex alignCenter justifyCenter bannerHeading",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Your Total Budget for",
                <br />,
                " ",
                "Kerala",
                " will be",
              ],
              "className": "f24 sfcw fw9 text-center ",
            },
            "ref": null,
            "rendered": Array [
              "Your Total Budget for",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              " ",
              "Kerala",
              " will be",
            ],
            "type": "p",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <BudgetRange
                priceMax={142767}
                priceMin={55152}
              />,
              <div
                className="absolute text-center p10 f12 sbc7 numberOfPeople"
              >
                for 
                1
                 
                adult
              </div>,
            ],
            "className": "absolute wfull budgetWrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "priceMax": 142767,
                "priceMin": 55152,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "for ",
                  1,
                  " ",
                  "adult",
                  null,
                ],
                "className": "absolute text-center p10 f12 sbc7 numberOfPeople",
              },
              "ref": null,
              "rendered": Array [
                "for ",
                1,
                " ",
                "adult",
                null,
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
